tests/integration_tests/functional/test_cmd_line_parameters.py

```diff
@@ -36,8 +36,10 @@ def test_describe_snapshot_all_versions(
     print(vm.log_data)
     vm.kill()
 
-    # Fetch Firecracker binary for the latest version
-    fc_binary, _ = get_firecracker_binaries()
+    # Starting from version 1.6.0, each Firecracker binary supports a (potentially)
+    # different snapshot data format version. So, here we use the binary that created
+    # the snapshot to check the --describe-snapshot command.
+    fc_binary = firecracker_release.path
```

Comment:
Problem is it can't. The snapshot header has changed in a way that it doesn't allow it parse older snapshot versions.

But you are right. I think the correct thing to do here is to skip this for older supported releases. WDYT?

Reply:
> I think the correct thing to do here is to skip this for older supported releases.

Yeah, this is what I meant, sorry :)

Reply:
Done :)

Reply:
Shouldn't the diff to the test be?

```diff
diff --git a/tests/integration_tests/functional/test_cmd_line_parameters.py b/tests/integration_tests/functional/test_cmd_line_parameters.py
index 59e26e4e9..99d17ca03 100644
--- a/tests/integration_tests/functional/test_cmd_line_parameters.py
+++ b/tests/integration_tests/functional/test_cmd_line_parameters.py
@@ -7,26 +7,22 @@ from pathlib import Path
 
 import pytest
 
-from framework.utils import run_cmd
+from framework.utils import run_cmd, get_firecracker_version_from_toml
 from host_tools.cargo_build import get_firecracker_binaries
 from host_tools.metrics import validate_fc_metrics
 
 
-def test_describe_snapshot_all_versions(
-    microvm_factory, guest_kernel, rootfs, firecracker_release
-):
+def test_describe_snapshot_all_versions(microvm_factory, guest_kernel, rootfs):
     """
     Test `--describe-snapshot` correctness for all snapshot versions.
 
     For each release create a snapshot and verify the data version of the
     snapshot state file.
     """
-    target_version = firecracker_release.snapshot_version
+    target_version = get_firecracker_version_from_toml()
     vm = microvm_factory.build(
         guest_kernel,
         rootfs,
-        fc_binary_path=firecracker_release.path,
-        jailer_binary_path=firecracker_release.jailer,
     )
     vm.spawn()
     vm.basic_config(track_dirty_pages=True)

```

Reply:
I _think_ not. IIUC the point of this test, it is to check if `firecracker --describe-snapshot` works correctly for all previously supported snapshot formats. Once we start having more Firecracker binaries that "speak" the new snapshot format this test will create a snapshot for every previously supported Firecracker version (>= 1.6.0) and then use the latest `main` to check that `--describe-snapshot` works for all of the previous snapshots.

Reply:
Ahhhh, I get it now. Didnt catch that it used the current firecracker version to call `--describe-snapshot`. Okay discard what I said :D